pkg/backup/handlers/arango/backup/arango_client_impl.go

```diff
@@ -128,9 +129,10 @@ func (ac *arangoClientBackupImpl) Get(backupID driver.BackupID) (driver.BackupMe
 	}
 }
 
-func (ac *arangoClientBackupImpl) getCredentialsFromSecret(secretName string) (interface{}, error) {
-
-	token, err := k8sutil.GetTokenSecret(ac.kubecli.CoreV1().Secrets(ac.backup.Namespace), secretName)
+func (ac *arangoClientBackupImpl) getCredentialsFromSecret(ctx context.Context, secretName string) (interface{}, error) {
+	ctxChild, cancel := context.WithTimeout(ctx, k8sutil.GetRequestTimeout())
+	token, err := k8sutil.GetTokenSecret(ctxChild, ac.kubecli.CoreV1().Secrets(ac.backup.Namespace), secretName)
+	cancel()
```

Comment:
Why not just use defer cancel() line above?

Reply:
yes, it is the place where `defer` can be used. My approach is:
- if the timeout is used only once then `defer` can be used
- otherwise cancel context immediately and reuse the same cancel function later in the same function